node-manager/src/seedgen.rs

```diff
@@ -1,7 +1,68 @@
+use bip32::XPrv;
 use bip39::Mnemonic;
+use bitcoin::secp256k1::{PublicKey, Secp256k1};
+use lightning::chain::keysinterface::{KeysInterface, KeysManager, Recipient};
 
 pub fn generate_seed() -> Mnemonic {
     let mut entropy = [0u8; 32];
     getrandom::getrandom(&mut entropy).expect("Failed to generate entropy");
     Mnemonic::from_entropy(&entropy).expect("Could not generate seed")
 }
+
+// A node private key will be derived from `m/0'/X'`, where it's node pubkey will
+// be derived from the LDK default being `m/0'/X'/0'`.
+pub fn derive_pubkey_child(mnemonic: Mnemonic, child_index: u32) -> PublicKey {
+    let xpriv = XPrv::new(&mnemonic.to_seed(""))
+        .unwrap()
+        .derive_child(bip32::ChildNumber::new(0, true).unwrap())
+        .unwrap()
+        .derive_child(bip32::ChildNumber::new(child_index, true).unwrap())
+        .unwrap();
+    let current_time = instant::now();
+    let keys_manager = KeysManager::new(
+        &xpriv.to_bytes(),
+        (current_time / 1000.0).round() as u64,
+        (current_time * 1000.0).round() as u32,
```

Comment:
this doesn't give us true nano seconds time, just milliseconds with extra 0s, we can just copy what sensei did here

https://github.com/L2-Technology/sensei/blob/3d76006d1a8509ac180886c057e281cc594fb545/senseicore/src/node.rs#L659

Reply:
seems `instant::now()` is just a layer on top of this anyways

Reply:
can't use `SystemTime::now()` in wasm and [instant](https://docs.rs/instant/0.1.12/instant/index.html) doesn't provide anything but millisecounds.